### PR TITLE
Fix cleanup_ci.py - fix cleanup of argo workflows and IAM policy bindings

### PR DESCRIPTION
**Pull request by jlewi**
Fix cleanup_ci.py - cleaning of Argo and service account binding

* Add logging to make it easier to monitor progress.
* Print out stack traces on errors.

* cleanup_workflows needs to get GKE credentials.
  * The project might be different from the project where KF is deployed.

Related to #358 cleanup ci not cleaning up Argo workflows

Related to #357 cleanup ci not cleaning up IAM policies.

* Fix bug in cleaning up Argo workflows; we should not break out of loop.

* Don't exit if delete of Workflow fails; it looks like there might be
  some race conditions.

* Update the docker image to use google-api-python-client library 1.7.0 instead
  of 1.6.5. It looks like using 1.6.5 gives us an error when getting IAM
  policy.

* Prune the OWNERs file to remove a bunch of approvers who haven't been active.

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/testing/359)

### CODE REVIEW

File: images/Dockerfile [hunk 1]
```diff
@@ -21,6 +21,7 @@ RUN set -ex \
     wget \
     ca-certificates \
     git \
+    emacs \
```
[the remark this answers is not in the document]
Comment (jlewi) [reply images/Dockerfile]: Its convenient for debugging; e.g. manually editing of files.

[the remark this answers is not in the document]
Comment (gabrielwen) [reply images/Dockerfile]: ah, ok.

**Pull request merged**